From 77df354fbaa5cd52413342a4af380641c8be0c65 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Thu, 30 Oct 2014 09:11:21 +0100
Subject: [PATCH 0005/1800] Bug 739375 - Crash when sort folder with Due By
 column ][

---
 e-util/e-table-extras.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/e-util/e-table-extras.c b/e-util/e-table-extras.c
index dcf15b0d02..43fca94435 100644
--- a/e-util/e-table-extras.c
+++ b/e-util/e-table-extras.c
@@ -117,7 +117,7 @@ e_int64ptr_compare (gconstpointer data1,
 		return (*pa == *pb) ? 0 : (*pa < *pb) ? -1 : 1;
 
 	/* sort unset values before set */
-	return pa ? 1 : -1;
+	return (!pa && !pb) ? 0 : (pa ? 1 : -1);
 }
 
 /* UTF-8 strncasecmp - not optimized */
-- 
2.15.0

