From 153d0babf10bb13b02e95921ed59eeb71aa6743f Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 24 Nov 2014 08:28:08 +0100
Subject: [PATCH 0100/1800] NEWS update for 3.13.8

---
 NEWS | 87 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 87 insertions(+)

diff --git a/NEWS b/NEWS
index 140dc07c79..fa763868c7 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,90 @@
+Evolution 3.13.8 2014-11-24
+---------------------------
+
+Bug Fixes:
+	Bug 300871 - Ignore a specific thread (Milan Crha)
+	Bug 545153 - Ask whether to quit with pending operations (Milan Crha)
+	Bug 739375 - Crash when sort folder with Due By column (Milan Crha)
+	Bug 739375 - Crash when sort folder with Due By column ][ (Milan Crha)
+	Bug 739364 - Drop references to geoclue (Milan Crha)
+	Bug 739569 - Reply joins lines incorrectly (Tomas Popela)
+	Bug 676471 - Double free when sorting by date columns in calendar (Milan Crha)
+	Bug 739577 - Runtime critical when saving message to drafts (Tomas Popela)
+	Bug 739659 - Composer has two different context menus (Tomas Popela)
+	Bug 739605 - Remove alert buttons on the alert bar hide (Milan Crha)
+	Bug 739562 - Message composer autosave not updating (Milan Crha)
+	Bug 739705 - The mail composer's attachments "toolbar" should have uniform spacing/padding (Tomas Popela)
+	Bug 739708 - Spell checking shows words with umlauts as false positives (Milan Crha)
+	Bug 739030 - Prevailing space in Subject column (Milan Crha)
+	Bug 739585 - [Automatic Contacts] Show configuration also in Edit->Plugins (Milan Crha)
+	Bug 657233 - Clean up "Calendar and Tasks" preferences in "General" and "Display" (Milan Crha)
+	Bug 709187 - Add an option to allow Expunge in vFolders (Milan Crha)
+	Bug 700778 - Correct TNEF mime type definition typo (David Girault)
+	Bug 736947 - Attachment icon not shown in Sent folder (Milan Crha)
+	Bug 736950 - Message with empty body and attachment shows body as attachment.dat (Milan Crha)
+	Bug 740410 - Paragraph keeps Preformatted after paste (Tomas Popela)
+	Bug 733518 - Make VOICE type optional for TEL in EContact (Milan Crha)
+	Bug 697858 - Slow message composer open (Milan Crha)
+
+Miscellaneous:
+	Avoid runtime warning when disabling collection ESource (Milan Crha)
+	Address two possible places causing runtime warning on a GSource removal (Milan Crha)
+	EHTMLEditor - Refactor code that sets the selection position into the elements (Tomas Popela)
+	EHTMLEditorSelection - Correctly save the selection markers in quoted content (Tomas Popela)
+	EHTMLEditorView - Run various functions only when the content of the composer is changed (Tomas Popela)
+	EHTMLEditorView - Correctly mark the paragraph that will receive input when the composer is opened (Tomas Popela)
+	EHTMLEditorView - Try to preserve the text blocks when processing the plain text content (Tomas Popela)
+	EHTMLEditorView - Optimize the put_body_in_citation function (Tomas Popela)
+	EHTMLEditorView - Decode the HTML encoded string only when ampersand is presented (Tomas Popela)
+	EHTMLEditorView - After wrapping use the faster approach for quoting the content (Tomas Popela)
+	EHTMLEditorView - Use BR element instead of zero width space character and selection markers instead of caret position (Tomas Popela)
+	[ECalOps] ECalModel not always provided when creating a new component editor (Milan Crha)
+	[EMailBackend] Provide cancellable and show in UI prepare for a quit (Milan Crha)
+	[Send through Outbox] Add option when to flush Outbox folder (Milan Crha)
+	EMailSignatureEditor, EHTMLEditorView - Don't insert unnecessary new line when editting the signature (Tomas Popela)
+	EHTMLEditorView - Don't use separate WebView for converting the HTML message to plain text, but do it in situ (Tomas Popela)
+	EHTMLEditorView - Queue the load operations if the view is not ready (Tomas Popela)
+	cal_comp_util_compare_event_timezones: Do not ask with NULL tzid (Milan Crha)
+	[Mark-All-Read] Save changes to the server immediately (Milan Crha)
+	EHTMLEditorView - Queue load operations for HTML content (Tomas Popela)
+	EmComposerUtils - Move the comment to the right place and remove code duplication (Tomas Popela)
+	EHTMLEditorView - Don't crash when the paste button is pressed and the view is not focused (Tomas Popela)
+	EHTMLEditor - Refactor the core editor actions (indent, unindent, ..) to their own group (Tomas Popela)
+	EMsgComposer - Make the undo and redo work in Subject and Reply-To input widgets (Tomas Popela)
+	EHTMLEditorActions - Process some of the EHTMLEditor actions just when is has focus (Tomas Popela)
+	Provide descriptive error message when an e-mail is not available in offline (Milan Crha)
+	Avoid automatic mail account update when the application is exiting (Milan Crha)
+	Revert "[EMailBackend] Provide cancellable and show in UI prepare for a quit" (Milan Crha)
+	EMsgComposer - Don't insert unnecessary new line into body, when forwarding an iCal event (Tomas Popela)
+	EHTMLEditorView - Fix logic that prevents the text blocks when processing the content into the composer (Tomas Popela)
+	Offer only enabled accounts as meeting organizers (Milan Crha)
+	Odd cross line when drawing an ECalendar with multiple columns (Milan Crha)
+	Share GSettings objects with e_util_ref_settings() (Milan Crha)
+	mail_regen_list: Fix a memory leak (Milan Crha)
+	Correct calendar Week/Month View event creation on double-click (Milan Crha)
+	GSettings Bindings rely on quarks (Milan Crha)
+	Commit initial Catalan help translation (Jordi Mas)
+	EHTMLEditorView - Improve detection of written e-mail addresses (Tomas Popela)
+	EHTMLEditorSelection - Don't leak font-color (Tomas Popela)
+	tests: contact editor: update lookup for email fields (Vadim Rutkovsky)
+	EHTMLEditorSelection - Don't leak background color and return default color values when in plain text mode (Tomas Popela)
+	tests: contact editor: don't collapse sections (Vadim Rutkovsky)
+	EHTMLEditorView - Use BR instead of zero width space when inserting new line into citation (Tomas Popela)
+	EHTMLEditorView - Get rid of GdkKeyEvent and use boolean flags instead. (Tomas Popela)
+	EHTMLEditorView - Remove unwanted attributes from HTML, HEAD and BODY elements when converting from HTML to plain text message (Tomas Popela)
+	[BBDB] Decode QP encoded names from recipients before adding them to a book (Milan Crha)
+	EHTMLEditorView - Improve detection of e-mail addresses in conversion from HTML to plain text (Tomas Popela)
+	EHTMLEditorView - Unref GRegex only if they are created (Tomas Popela)
+	EHTMLEditorView - Fix two memory leaks (Milan Crha)
+
+Translations:
+	Dimitris Spingos (el)
+	Timo Jyrinki (fi)
+	Daniel Mustieles (es)
+	Kjartan Maraas (nb)
+	Duarte Loreto (pt)
+	Dimitris Spingos (el)
+
 Evolution 3.13.7 2014-10-27
 ---------------------------
 
-- 
2.15.0

