From 06088f173c51d890371b18b57be06e2672919129 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 24 Nov 2014 12:23:47 +0100
Subject: [PATCH 0102/1800] Bug 740197 - Month view doesn't always indicate
 excess appointments

---
 libgnomecanvas/gnome-canvas-pixbuf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libgnomecanvas/gnome-canvas-pixbuf.c b/libgnomecanvas/gnome-canvas-pixbuf.c
index 125089754c..87c3c26c3b 100644
--- a/libgnomecanvas/gnome-canvas-pixbuf.c
+++ b/libgnomecanvas/gnome-canvas-pixbuf.c
@@ -272,6 +272,9 @@ gnome_canvas_pixbuf_draw (GnomeCanvasItem *item,
 
 	gnome_canvas_item_i2c_matrix (item, &matrix);
 
+	matrix.x0 -= x;
+	matrix.y0 -= y;
+
 	cairo_save (cr);
 	cairo_transform (cr, &matrix);
 
-- 
2.15.0

