From 18413e1a416d6ad0ee50825293d4ca11029ce8a3 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 26 Jan 2015 09:25:45 +0100
Subject: [PATCH 0193/1800] Post-release version bump

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ef573149dd..49e4547366 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,7 +1,7 @@
 dnl Evolution Versions
 m4_define([evo_major_version], [3])
 m4_define([evo_minor_version], [13])
-m4_define([evo_micro_version], [10])
+m4_define([evo_micro_version], [90])
 m4_define([evo_version],
 	[evo_major_version.evo_minor_version.evo_micro_version])
 m4_define([evo_stable_version],
-- 
2.15.0

