From 7637535cba9890a826d02c59c9de88caa7422c16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20=C4=8Cernock=C3=BD?= <marek@manet.cz>
Date: Thu, 5 Feb 2015 15:08:07 +0100
Subject: [PATCH 0255/1800] Fixed plural forms handling

---
 e-util/e-html-editor-replace-dialog.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/e-util/e-html-editor-replace-dialog.c b/e-util/e-html-editor-replace-dialog.c
index edc0489786..5f0113f51e 100644
--- a/e-util/e-html-editor-replace-dialog.c
+++ b/e-util/e-html-editor-replace-dialog.c
@@ -111,7 +111,10 @@ webkit_find_controller_counted_matches_cb (WebKitFindController *find_controller
 			selection, TRUE, E_HTML_EDITOR_SELECTION_GRANULARITY_WORD);*/
 	}
 
-	result = g_strdup_printf (_("%d occurences replaced"), match_count);
+	result = g_strdup_printf (ngettext("%d occurence replaced",
+					   "%d occurences replaced",
+					   match_count),
+				 match_count);
 	gtk_label_set_label (GTK_LABEL (dialog->priv->result_label), result);
 	gtk_widget_show (dialog->priv->result_label);
 	g_free (result);
-- 
2.15.0

