From 57261c1f51f2f9ed8aaa43848decea8601b3c4df Mon Sep 17 00:00:00 2001
From: Andre Klapper <ak-47@gmx.net>
Date: Mon, 16 Feb 2015 00:09:21 +0100
Subject: [PATCH 0298/1800] Update Spam settings (UI offers less options)

---
 help/C/mail-spam-settings.page | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/help/C/mail-spam-settings.page b/help/C/mail-spam-settings.page
index 213545d8cd..86f1243ecb 100644
--- a/help/C/mail-spam-settings.page
+++ b/help/C/mail-spam-settings.page
@@ -9,7 +9,7 @@
     <link type="guide" xref="mail-spam" />
     <link type="seealso" xref="mail-spam-marking"/>
 
-    <revision pkgversion="3.8.1" version="0.9" date="2013-06-22" status="final"/>
+    <revision pkgversion="3.12.9" version="0.10" date="2015-03-15" status="final"/>
     <credit type="author">
       <name its:translate="no">Andre Klapper</name>
       <email its:translate="no">ak-47@gmx.net</email>
@@ -66,12 +66,10 @@
 <item><p>Select <gui>Mail Preferences</gui>.</p></item>
 <item><p>Click on the <gui>Junk</gui> tab. Here, you can specify the following:</p>
 <list>
-<item><p>Checking incoming mail for junk messages.</p></item>
+<item><p>Checking incoming messages for junk.</p></item>
 <item><p>Deleting junk messages upon exit, and how often junk messages are to be deleted.</p></item>
-<item><p>Checking custom headers for junk.</p></item>
+<item><p>Checking custom mail headers (added by your mail server) for junk.</p></item>
 <item><p>Marking messages as not junk if the sender is in the address book.</p></item>
-<item><p>The default junk filter.</p></item>
-<item><p>Bogofilter and SpamAssassin options.</p></item>
 </list>
 </item>
 </steps>
-- 
2.15.0

