From b57d81c1f0a5e110efc61356bbb918b8a01c2f30 Mon Sep 17 00:00:00 2001
From: Andre Klapper <ak-47@gmx.net>
Date: Mon, 16 Feb 2015 00:26:41 +0100
Subject: [PATCH 0305/1800] Order of items in UI has changed

Bug 731787#c1
---
 help/C/mail-spam-settings.page | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/help/C/mail-spam-settings.page b/help/C/mail-spam-settings.page
index 86f1243ecb..b99b0268bf 100644
--- a/help/C/mail-spam-settings.page
+++ b/help/C/mail-spam-settings.page
@@ -9,7 +9,7 @@
     <link type="guide" xref="mail-spam" />
     <link type="seealso" xref="mail-spam-marking"/>
 
-    <revision pkgversion="3.12.9" version="0.10" date="2015-03-15" status="final"/>
+    <revision pkgversion="3.12.9" version="0.11" date="2015-03-15" status="final"/>
     <credit type="author">
       <name its:translate="no">Andre Klapper</name>
       <email its:translate="no">ak-47@gmx.net</email>
@@ -68,8 +68,8 @@
 <list>
 <item><p>Checking incoming messages for junk.</p></item>
 <item><p>Deleting junk messages upon exit, and how often junk messages are to be deleted.</p></item>
-<item><p>Checking custom mail headers (added by your mail server) for junk.</p></item>
 <item><p>Marking messages as not junk if the sender is in the address book.</p></item>
+<item><p>Checking custom mail headers (added by your mail server) for junk.</p></item>
 </list>
 </item>
 </steps>
-- 
2.15.0

