From 5369567c57e5feae144809e031a2256f44d04669 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 16 Feb 2015 09:14:41 +0100
Subject: [PATCH 0309/1800] NEWS update for 3.13.90

---
 NEWS | 103 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 103 insertions(+)

diff --git a/NEWS b/NEWS
index 012488899d..1a8d01dcc5 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,106 @@
+Evolution 3.13.90 2015-02-16
+----------------------------
+
+Bug Fixes:
+	Bug 742611 - Extra empty line added when deleting sections from quoted text when replying to a mail (Tomas Popela)
+	Bug 743288 - Extra space left when breaking a long paragraph (Tomas Popela)
+	Bug 743550 - Invalid free in e_html_editor_view_add_inline_image_from_element() (Milan Crha)
+	Bug 743959 - Automatically collapsing threads after folder update (Milan Crha)
+	Bug 743876 - Inline image lost on message send when sending through Outbox (Tomas Popela)
+	Bug 743875 - Invalid read on message send (Tomas Popela)
+	Bug 249749 - Threaded Message List Expands When Using "Find Now" (Milan Crha)
+	Bug 686620 - Change wording of a New Account autodiscovery option (Milan Crha)
+	Bug 743109 - Do not require spam software during build time (Milan Crha)
+	Bug 606239 - Better indication of signed and/or encrypted received email (Christian Schaarschmidt)
+	Bug 731502 - Import mails and contacts from KMail (David Liang)
+	Bug 744218 - Cannot move with arrow keys from top to bottom (Tomas Popela)
+	Bug 743942 - Disambiguate string "Insert Image" (Milan Crha)
+	Bug 737018 - GUI string for translation "New" without context (Milan Crha)
+	Bug 743547 - Report one error only once for folders refresh (Milan Crha)
+	Bug 744372 - When pasting text evolution drops the carriage returns (Tomas Popela)
+	Bug 744373 - Delete whole line with selection lefts empty line (Tomas Popela)
+	Bug 744363 - Indefinite form used for specific objects in error messages (Milan Crha)
+	Bug 732999 - Middle-click paste doesn't move cursor (Tomas Popela)
+	Bug 741857 - Doesn't close after mailto: message sent (Milan Crha)
+
+Miscellaneous:
+	EHTMLEditorView - Spell check is sometimes lost when deleting content (Tomas Popela)
+	EMailFormatter - Fix the debug output (Tomas Popela)
+	Avoid/workaround usage of an EAsyncClosure when opening an EClient (Milan Crha)
+	EHTMLEditorSelection - Save the selection marks to right position when it starts or ends near the quote marks (Tomas Popela)
+	Fix various problems with deleting the selection that ends or starts in the quoted content (Tomas Popela)
+	Rework how the selection inside the qouted content is deleted. (Tomas Popela)
+	EHTMLEditorView - Block is left in quoted content when deleting the last in quoted content when signature is presented (Tomas Popela)
+	EHTMLEditorView - Don't insert new line when we hit the empty paragraph (Tomas Popela)
+	Updated German translation (Christian Kirbach)
+	EMComposerUtils - Drop unused prefix and suffix around the message (Tomas Popela)
+	EMailFormatterQuote - Close the span element instead of wrongly creating the new one (Tomas Popela)
+	Message credits are put on the end of quoted content instead of on its beginning. (Tomas Popela)
+	EHTMLEditorView - When there is no block in BODY, create the default one (Tomas Popela)
+	Move authentication of backends back to the client (Milan Crha)
+	EHTMLEditorView - Remove meaningless comment (Tomas Popela)
+	Replace _BSD_SOURCE with _DEFAULT_SOURCE define (Milan Crha)
+	Mute "Shell not finalized on exit" runtime warning (Milan Crha)
+	Simplify code at shell_trust_prompt_done_cb() a bit (Milan Crha)
+	EHTMLEditorView - Don't send the Evolution CSS style element in HEAD with HTML messages (Tomas Popela)
+	EHTMLEditorView - Remove the node that is used for quoting when whe quote the body (Tomas Popela)
+	EHTMLEditorView - Style that is used to make the links clickable is send when using Ctrl + Enter to send the message (Tomas Popela)
+	EHTMLEditorView - Don't send the selection markers when sending a HTML mail (Tomas Popela)
+	EHTMLEditorView - Remove the Draft style sheet when the draft is loaded (Tomas Popela)
+	EWebView, EHTMLEditorView - Improve citations look (Tomas Popela)
+	EWebView - Don't show unnecessary white space around pre and code elements (Tomas Popela)
+	Avoid busy loop on Calendar's Month view select (Milan Crha)
+	EHTMLEditorSelection - Correctly wrap after link that does not fit on the line (Tomas Popela)
+	EWebView - When showing plain text template from editor in the preview it shows the citation lines as well as ">" (Tomas Popela)
+	Move the comment about the citation level colors to the right place (Tomas Popela)
+	EHTMLEditorView - Try to convert the content only, when we load something (Tomas Popela)
+	EHTMLEditorView - Insert new paragraph when we loaded an empty string or HTML (Tomas Popela)
+	EHTMLEditorView - The links in composer are clickable by default (Tomas Popela)
+	Created active link for grep manpages (Marek Černocký)
+	Fixed tag type (Marek Černocký)
+	Fixed plural forms handling (Marek Černocký)
+	Ignore errors when searching for email address in books (Milan Crha)
+	Use automake subdir-objects option (Milan Crha)
+	Remove "Unknown background operation" generic job description message (Milan Crha)
+	EHTMLEditorSelection - Non collapsed selection is collapsed after saving its start and end points (Tomas Popela)
+	EHTMLEditorView - Unnecessary new line inserted when composer when creating new message (Tomas Popela)
+	EHTMLEditorView - Replace the active selection when inserting emoticons (Tomas Popela)
+	EHTMLEditorSelection - Replace active selection when inserting inline image into composer (Tomas Popela)
+	Prefer g_output_stream_write_all() over g_output_stream_write() (Milan Crha)
+	Postscript attachments not shown in message preview (Milan Crha)
+	Composer - Make drag and drop usable (Tomas Popela)
+	EHTMLEditorView, EHTMLEditorSelection - Use webkit_dom_html_element_insert_adjacent_html (Tomas Popela)
+	EHTMLEditorSelection - Don't treat links as underlined (Tomas Popela)
+	Use camel_application_is_exiting declaration from camel.h (Milan Crha)
+	EHTMLEditorView - Use webkit_dom_html_element_insert_adjacent_html in parse_html_into_paragraphs (Tomas Popela)
+	S/MIME mail parser not used, but compiled (Milan Crha)
+	Introduce e_html_editor_selection_set_on_point and use it where possible (Tomas Popela)
+	Switch side of a wider border for signed/encrypted mail frame (Milan Crha)
+	[Win32] Be able to load plugins by providing correct build-time paths (Milan Crha)
+	[Win32] Update gtk patch to preset "always-show-image" to TRUE (Milan Crha)
+	Cover how to force online mode via GIO_USE_NETWORK_MONITOR (Andre Klapper)
+	Update Spam settings (UI offers less options) (Andre Klapper)
+	Small string change in UI (Andre Klapper)
+	Use <gui> markup (Andre Klapper)
+	Remove non-existing UI options (Andre Klapper)
+	Improve explanation of further available account options (Andre Klapper)
+	Cover the "concurrent connections" setting (Andre Klapper)
+	Fix typo in Junk Testing Settings UI (Andre Klapper)
+	Order of items in UI has changed (Andre Klapper)
+	Drop all that geeky terminal commands and link to Seahorse (Andre Klapper)
+	Cover why mail signatures start with "-- " (Andre Klapper)
+	Remove doc references to EMailAuthenticator (Milan Crha)
+
+Translations:
+	Christian Kirbach (de)
+	Sveinn í Felli (is)
+	Marek Černocký (cs)
+	Daniel Mustieles (es)
+	Dušan Kazik (sk)
+	Kjartan Maraas (nb)
+	Piotr Drąg (pl)
+	Balázs Úr (hu)
+
 Evolution 3.13.10 2015-01-26
 ----------------------------
 
-- 
2.15.0

