From 9fca49c1da2f3b6253cf7bfa4865e5bd61a837a7 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 2 Mar 2015 08:50:46 +0100
Subject: [PATCH 0394/1800] NEWS update for 3.15.91

---
 NEWS | 80 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 80 insertions(+)

diff --git a/NEWS b/NEWS
index 1a8d01dcc5..4d7009baeb 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,83 @@
+Evolution 3.15.91 2015-03-02
+----------------------------
+
+Bug Fixes:
+	Bug 744471 - Correct may/might usage in gschema strings (Milan Crha)
+	Bug 744541 - No plural use in KMail importer (David Liang)
+	Bug 744444 - Typo (encrpyted) in translatable strings (Milan Crha)
+	Bug 681877 - tzset() called too many times (Milan Crha)
+	Bug 744550 - Always scroll to selected message after messages delete (Milan Crha)
+	Bug 743410 - White-spaces lost in plain text message printouts (Milan Crha)
+	Bug 741093 - Force icon sizes when loading theme icons (Milan Crha)
+	Bug 740169 - Add support for smileys via Unicode-characters (Jonas Hahnfeld)
+	Bug 744562 - Allow insert of emoticons in plain text mode (Jonas Hahnfeld)
+	Bug 619449 - Preselect default source for Task/Memo panes in Calendar view (Milan Crha)
+	Bug 744401 - Many WebKitDom objects leaked for every message viewed (Tomas Popela)
+	Bug 744401 - Many WebKitDom objects leaked for every message viewed (Tomas Popela)
+	Bug 741013 - Pass URIs into the primary instance as soon as possible (Milan Crha)
+	Bug 744849 - Insert->Emoticon deletes text before cursor (Tomas Popela)
+	Bug 744848 - Text lost on format change (Tomas Popela)
+	Bug 745013 - handle_http_request: GBinding not thread safe (Milan Crha)
+	Bug 743404 - Status column of message list exposed poorly via atk (Mike Gorse)
+	Bug 237917 - Arbitrary entry in From: field when composing (Valentin Blot)
+	Bug 745044 - Missing mnemonics in backup-restore plugin (Milan Crha)
+	Bug 745029 - Wrong calendar used for .ics file import (Milan Crha)
+	Bug 737300 - Make message list header smaller (Milan Crha)
+	Bug 737300 - Make message list header draw closer to GtkTreeView (Milan Crha)
+	Bug 738080 - message/rfc822 attachment drag and drop often does not work (Milan Crha)
+	Bug 621751 - Allow import of non-MBOX format mail messages (Milan Crha)
+
+Miscellaneous:
+	Avoid crash when validating a new filter rule (Milan Crha)
+	Updated Spanish translation (Daniel Mustieles)
+	Use CalDAV to get list of available Google calendars (Milan Crha)
+	Special-case Google 'caldav' calendars to use 'google' editor (Milan Crha)
+	Remove declaration of nonexistent e_cal_model_get_extension_name() (Milan Crha)
+	EHTMLEditorSelection - Add missing colon in properties documentation (Tomas Popela)
+	EHTMLEditorSelection - Don't unref the EHTMLEditorView when it is used afterwards (Tomas Popela)
+	EHTMLEditorSelection - The font color for selected text is wrong (Tomas Popela)
+	EHTMLEditorSelection - After settings for size to some value and back to the default the font is a little bit smaller (Tomas Popela)
+	EHTMLEditorSelection - Spell check is lost after some font style operations (Tomas Popela)
+	EHTMLEditorSelection - Try to remove the link only when we find anchor (Tomas Popela)
+	EHTMLEditorSelection - Don't unref the EHTMLEditorView object when it is used afterwards (Tomas Popela)
+	EHTMLEditorSelection - Rework how the monospaced style is set (Tomas Popela)
+	EHTMLEditorSelection - Use create_selection_marker function when saving selection (Tomas Popela)
+	Simplify e-web-view.c:add_css_rule_into_style_sheet() (Milan Crha)
+	handle_http_request: Skip invalid URIs (Milan Crha)
+	Use a secure URL for queries from gravatar.com (Milan Crha)
+	Address few more WebKitDOM objects memory leaks (Milan Crha)
+	MessageList - Localize global string arrays only once (Milan Crha)
+	Use thread safe variants of g_object_bind_property*() functions from evolution-data-server (Milan Crha)
+	Update devel-doc Since from 3.14 to 3.16 (Milan Crha)
+	itip_send_comp_sync: Initialize all members of a stack allocated structure (Milan Crha)
+	Fix a crash when 'Check for supported authentication types' clicked (Milan Crha)
+	[ETree/ETable] Implement GtkScrollableInterface::get_border() for Gtk 3.15.9+ (Milan Crha)
+	HTML events callbacks are not called (Tomas Popela)
+	EMailDisplay - Fix more leaks of WebKit objects (Tomas Popela)
+	EMailReader - Fix more leaks of WebKit objects (Tomas Popela)
+	ITIP - Fix more leaks of WebKit objects (Tomas Popela)
+	EHTMLEditorActions - Activating the Paste as Text and Paste Quotation from context menu does nothing (Tomas Popela)
+	EHTMLEditorView - When inserting inline image refresh spell check just in current block (Tomas Popela)
+	EHTMLEditorSelection - Use variable that holds if the range is collapsed (Tomas Popela)
+	EHTMLEditorView - Emoticons can be resizable just in HTML mode (Tomas Popela)
+	EHTMLEditorView - Fix the Past Quotation action (Tomas Popela)
+	EHTMLEditorView - Be more efficient when trying to get if we are going to remove last element in body (Tomas Popela)
+	EHTMLEditorView - Block is wrongly preserved in the beginning of quoted content (Tomas Popela)
+	EHTMLEditorView - When pasting text with space on end ends with space lost (Tomas Popela)
+	EWebView - Fix wrong indentation (Tomas Popela)
+	Sync version with GNOME 3.15 series (Milan Crha)
+
+Translations:
+	Kjartan Maraas (nb)
+	Daniel Mustieles (es)
+	Muhammet Kara (tr)
+	Chao-Hsiung Liao (zh_TW)
+	Dušan Kazik (sk)
+	Dimitris Spingos (el)
+	Efstathios Iosifidis (el)
+	Balázs Úr (hu)
+	Maria  Mavridou (el)
+
 Evolution 3.13.90 2015-02-16
 ----------------------------
 
-- 
2.15.0

