From 39ede8e29eb747e41b5ce863dfc75c823e6b6166 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Tue, 3 Mar 2015 19:08:28 +0100
Subject: [PATCH 0404/1800] Bug 745338 - Disambiguate string "_Replace"

---
 e-util/e-html-editor-replace-dialog.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/e-util/e-html-editor-replace-dialog.c b/e-util/e-html-editor-replace-dialog.c
index 5f0113f51e..ea6870a6a4 100644
--- a/e-util/e-html-editor-replace-dialog.c
+++ b/e-util/e-html-editor-replace-dialog.c
@@ -380,7 +380,7 @@ e_html_editor_replace_dialog_new (EHTMLEditor *editor)
 			"editor", editor,
 			"icon-name", "edit-find-replace",
 			"resizable", FALSE,
-			"title", N_("Replace"),
+			"title", C_("dialog-title", "Replace"),
 			"transient-for", gtk_widget_get_toplevel (GTK_WIDGET (editor)),
 			"type", GTK_WINDOW_TOPLEVEL,
 			"window-position", GTK_WIN_POS_CENTER_ON_PARENT,
-- 
2.15.0

