From 7d227c0b8e6f2658f80fd628b9b1bc5b3c0adf49 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 4 Mar 2015 09:42:56 +0100
Subject: [PATCH 0408/1800] Bug 736432 - Attachment rename not shown in UI

---
 e-util/e-attachment.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/e-util/e-attachment.c b/e-util/e-attachment.c
index eb74037c09..bce35c6e7e 100644
--- a/e-util/e-attachment.c
+++ b/e-util/e-attachment.c
@@ -1090,11 +1090,11 @@ e_attachment_init (EAttachment *attachment)
 		attachment, "notify::encrypted",
 		G_CALLBACK (attachment_update_icon_column), NULL);
 
-	e_signal_connect_notify (
+	g_signal_connect (
 		attachment, "notify::file-info",
 		G_CALLBACK (attachment_update_file_info_columns), NULL);
 
-	e_signal_connect_notify (
+	g_signal_connect (
 		attachment, "notify::file-info",
 		G_CALLBACK (attachment_update_icon_column), NULL);
 
-- 
2.15.0

