From 6c879e8adcad1a46f1a9f765d1697e4475d74bec Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 4 Mar 2015 18:47:14 +0100
Subject: [PATCH 0415/1800] Bug 735956 - [EWebView] Do not scroll and zoom when
 Ctrl+mouse-wheel

---
 e-util/e-web-view.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/e-util/e-web-view.c b/e-util/e-web-view.c
index fa67d5f7a8..21625c682d 100644
--- a/e-util/e-web-view.c
+++ b/e-util/e-web-view.c
@@ -1021,6 +1021,8 @@ web_view_scroll_event (GtkWidget *widget,
 			} else if (total_delta_y <= -1.0) {
 				total_delta_y = 0.0;
 				direction = GDK_SCROLL_UP;
+			} else if (event->delta_y >= 1e-9 || event->delta_y <= -1e-9) {
+				return TRUE;
 			} else {
 				return FALSE;
 			}
-- 
2.15.0

