From 24bc2b08311f2d239079c47e6baf1fb637b2f714 Mon Sep 17 00:00:00 2001
From: Andre Klapper <a9016009@gmx.de>
Date: Mon, 9 Mar 2015 00:43:25 +0100
Subject: [PATCH 0443/1800] Split a string so we can reuse it in a different
 page.

---
 help/C/contacts-ldap.page | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/help/C/contacts-ldap.page b/help/C/contacts-ldap.page
index afe3c9915f..54516cf4d3 100644
--- a/help/C/contacts-ldap.page
+++ b/help/C/contacts-ldap.page
@@ -5,7 +5,7 @@
   <info>
     <link type="guide" xref="contacts-using-several-addressbooks"/>
 
-    <revision pkgversion="3.1.5" version="0.3" date="2011-07-28" status="final"/>
+    <revision pkgversion="3.1.5" version="0.4" date="2015-03-09" status="final"/>
 
     <credit type="author">
       <name its:translate="no">Andre Klapper</name>
@@ -67,8 +67,8 @@
 	  correct settings.</p>
         </item>
         <item>
-	  <p>Search Scope: The search scope is the breadth of a given search.
-	  The following options are available:</p>
+	  <p>Search Scope: The search scope is the breadth of a given search.</p>
+	  <p>The following options are available:</p>
           <list>
             <item>
 	      <p>One: Searches the Search Base and one entry below it.</p>
-- 
2.15.0

