From e3b955fefb0ada2a5673f4bd0d9637274e77529b Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 11 Mar 2015 15:25:30 +0100
Subject: [PATCH 0494/1800] Bug 746031 - Add missing library to composer build
 options

---
 composer/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/composer/Makefile.am b/composer/Makefile.am
index 86a411a07e..b0db589c1b 100644
--- a/composer/Makefile.am
+++ b/composer/Makefile.am
@@ -56,6 +56,7 @@ libevolution_mail_composer_la_LDFLAGS = -avoid-version $(NO_UNDEFINED) $(CODE_CO
 libevolution_mail_composer_la_LIBADD =			\
 	$(top_builddir)/e-util/libevolution-util.la	\
 	$(top_builddir)/shell/libevolution-shell.la	\
+	$(top_builddir)/libemail-engine/libemail-engine.la	\
 	$(top_builddir)/em-format/libevolution-mail-formatter.la \
 	$(top_builddir)/addressbook/gui/contact-editor/libecontacteditor.la		\
 	$(top_builddir)/addressbook/gui/contact-list-editor/libecontactlisteditor.la	\
-- 
2.15.0

