From 4e08c1a2d5981d0a0ccd7bdf8bf1471ec5823975 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Fri, 13 Mar 2015 09:46:46 +0100
Subject: [PATCH 0514/1800] Also check for 'help' uri scheme handler
 availability

---
 e-util/e-misc-utils.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/e-util/e-misc-utils.c b/e-util/e-misc-utils.c
index cfe9dfff99..82b436918f 100644
--- a/e-util/e-misc-utils.c
+++ b/e-util/e-misc-utils.c
@@ -270,6 +270,16 @@ e_misc_utils_is_help_package_installed (void)
 
 	g_free (path);
 
+	if (is_installed) {
+		GAppInfo *help_handler;
+
+		help_handler = g_app_info_get_default_for_uri_scheme ("help");
+
+		is_installed = help_handler && g_app_info_get_commandline (help_handler);
+
+		g_clear_object (&help_handler);
+	}
+
 	return is_installed;
 }
 
-- 
2.15.0

