From 4b896f611be765af3c88d839529b831fe792b608 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 23 Mar 2015 08:10:18 +0100
Subject: [PATCH 0549/1800] NEWS update for 3.16.0

---
 ChangeLog |  2 +-
 NEWS      | 12 ++++++++++++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index c26f1f78cf..69f1a6705f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -4,4 +4,4 @@
 	Get a clone of git repository and list changes with 'git log' there,
 	   $ git clone git://git.gnome.org/evolution
 	or browse changes online at
-	   http://git.gnome.org/browse/evolution/log/?h=master
+	   http://git.gnome.org/browse/evolution/log/?h=gnome-3-16
diff --git a/NEWS b/NEWS
index c950eb40d9..5ca26be9d2 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,15 @@
+Evolution 3.16.0 2015-03-23
+---------------------------
+
+Translations:
+	Chao-Hsiung Liao (zh_TW)
+	Josef Andersson (sv)
+	Christian Kirbach (de)
+	Claude Paroz (fr)
+	Daniel Mustieles (es)
+	Gábor Kelemen (hu)
+	Dimitris Spingos (el)
+
 Evolution 3.15.92 2015-03-16
 ----------------------------
 
-- 
2.15.0

