From d2247c06304758d93936ff589bd8fea84600e85c Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 25 May 2015 08:23:05 +0200
Subject: [PATCH 0763/1800] NEWS update for 3.17.2

---
 NEWS | 67 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 67 insertions(+)

diff --git a/NEWS b/NEWS
index b9c05b9f22..774504afa2 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,70 @@
+Evolution 3.17.2 2015-05-25
+---------------------------
+
+Bug Fixes:
+	Bug 748359 - To header collapsed in printed version when many addressees (Milan Crha)
+	Bug 707356 - Move folder confirmation dialog needs redone (Milan Crha)
+	Bug 707521 - Editing detached instance in editor removes Recurrence ID (Milan Crha)
+	Bug 747294 - Typos in documentation (Anders Jonsson)
+	Bug 668808 - Can't cancel mbox file import (Milan Crha)
+	Bug 672129 - Content-Disposition header confuses mail parser for text/plain (Milan Crha)
+	Bug 748679 - Cannot middle-click paste text from composer (Tomas Popela)
+	Bug 704824 - Dropdown list of calendars in ITIP mail becomes empty after accepting appointment (Tomas Popela)
+	Bug 671546 - Search with 'contains' in composer auto-completion (Milan Crha)
+	Bug 674927 - Add an option for default snooze time interval (Milan Crha)
+	Bug 748774 - Act on encrypt-by-default from ESourceOpenPGP (Daniel Kahn Gillmor)
+	Bug 748878 - Crash in ect_dispose() on quit (Milan Crha)
+	Bug 749217 - Backspace unindents multiple pasted paragraphs (Tomas Popela)
+	Bug 749060 - Help->Contents does not open in yelp (Milan Crha)
+	Bug 748574 - Upgrade to 3.16.1.1 from 3.12.11 has broken my email signature images (Tomas Popela)
+	Bug 749108 - Embedded inline images pretend message attachments (Milan Crha)
+	Bug 531054 - Reply to All changes To: and Cc: (Milan Crha)
+	Bug 593272 - Message preview shows Reply-to header as is (Milan Crha)
+	Bug 748981 - Option to always show sign/encrypt toolbar buttons in composer (Milan Crha)
+	Bug 748683 - Unhelpful 'Failed to connect' message (Milan Crha)
+	Bug 749613 - Crash when trying to wrap non-existing node (Tomas Popela)
+	Bug 749579 - Whitespace corrupted in outbound email (Tomas Popela)
+
+Miscellaneous:
+	Remove the last leftover of an e-cell-renderer-color gtk-doc reference (Milan Crha)
+	EMailRequest -  SVG images in HTML email are not displayed (Tomas Popela)
+	EHTMLEditorView - Runtime warning when editing HTML message as new (Tomas Popela)
+	EHTMLEditorView - Remove unneeded space (Tomas Popela)
+	EHTMLEditorView - Fix the 'Paste Quotation' action (Tomas Popela)
+	EHTMLEditorView - New line from the beginning of pasted text lost when pasting (Tomas Popela)
+	EHTMLEditorView - Preformated block in quoted content is wrapped when splitted with new line (Tomas Popela)
+	Composer - Insert credits, headers and other information in Normal block (Tomas Popela)
+	Misleading counts of events in the Calendar view's mini-calendar (Milan Crha)
+	EHTMLEditorActions - Optimize history saving for 'Cut' action (Tomas Popela)
+	EHTMLEditorView - Minor code and style changes (Tomas Popela)
+	Attachment load/save should keep a reference to its top-level window (Milan Crha)
+	Updated Czech translation (Marek Černocký)
+	EHTMLEditorView - Connect the 'changed::antialiasing' signal to the right object (Tomas Popela)
+	EHTMLEditorSelection - Fix text wrapping (Tomas Popela)
+	EHTTPRequest - GFileInfo not valid when machine is under heavy load (Tomas Popela)
+	EHTMLEditorSelection - Don't wrap every block in selection during 'Wrap lines' action (Tomas Popela)
+	Composer - Undo initiated with shortcuts doesn't stop (Tomas Popela)
+	Composer - Undo/Redo on operations with selection could be wrong (Tomas Popela)
+	EHTMLEditorView - Don't print coordinates for HISTORY_AND and HISTORY_START events (Tomas Popela)
+	EHTMLEditorView - Replace g_timeout_add with g_timeout_add_seconds (Tomas Popela)
+	EHTMLEditorSelection - Clean the code used for wrapping (Tomas Popela)
+	EHTMLEditorSelection - Selection markers are saved on wrong position after Tabulator (Tomas Popela)
+	EHTMLEditorView - Fix possible leak (Tomas Popela)
+	ExternalEditor - Fix leaks of WebKit objects (Tomas Popela)
+	Correct signature stripping in HTML messages (Milan Crha)
+	EHTMLEditorView - Busy loop after writing ";)" into composer (Tomas Popela)
+	Address some of the Coverity Scan issues (Milan Crha)
+	Fix a double-free on a picture gallery dispose (composer close) (Milan Crha)
+
+Translations:
+	Marek Černocký (cs)
+	Sveinn í Felli (is)
+	Kjartan Maraas (nb)
+	Piotr Drąg (pl)
+	Matej Urbančič (sl)
+	Dušan Kazik (sk)
+	Daniel Mustieles (es)
+
 Evolution 3.17.1 2015-04-27
 ---------------------------
 
-- 
2.15.0

