From 0168db11e53bce7b4e005e5acbf4ab823d92b123 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 20 Jul 2015 08:17:26 +0200
Subject: [PATCH 0896/1800] NEWS update for 3.17.4

---
 NEWS | 71 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 71 insertions(+)

diff --git a/NEWS b/NEWS
index 61253a3c4f..d1321a5448 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,74 @@
+Evolution 3.17.4 2015-07-20
+---------------------------
+
+Bug Fixes:
+	Bug 701483 - Inline GPG decryption sometimes hides the decrypted text (Milan Crha)
+	Bug 750615 - Using Ctrl+Z in plain text composer messes up content (Tomas Popela)
+	Bug 377352 - Per-folder setting to auto-archive messages based on their age (Milan Crha)
+	Bug 510970 - [itip] Let user select calendar even when meeting found (Milan Crha)
+	Bug 751194 - Bulleted lists indentation jumps levels (Tomas Popela)
+	Bug 751424 - [WeekView] Location added to Summary with inline editing (Milan Crha)
+	Bug 751488 - Add Birthday and Anniversary columns into Contacts view (Milan Crha)
+	Bug 751594 - Cannot Tab through whole Composer General preferences (Milan Crha)
+	Bug 751620 - OK button disabled by default in Copy to Calendar dialogue (Milan Crha)
+	Bug 680451 - Add Certificate support into Contact editor (Milan Crha)
+	Bug 751225 - Spell check selection changes font settings to initial settings (Tomas Popela)
+	Bug 751186 - Tab not part of editing undo / redo stack and corrupts the stack (Tomas Popela)
+	Bug 751915 - Crash when using both internal and external editors to compose plain-text mail (Tomas Popela)
+	Bug 751943 - Correct typo 'Teusday' (Milan Crha)
+	Bug 752018 - Events at midnight and 0 minutes duration hidden in UI (Milan Crha)
+	Bug 752119 - Avoid usage of gtk_style_context_get_background_color() (Milan Crha)
+	Bug 752086 - Paste with a space on the right adds also new-line (Tomas Popela)
+	Bug 751899 - Inconsistent form of words describing certificate trust (Milan Crha)
+	Bug 752199 - Do not require libgdata (Milan Crha)
+	Bug 724017 - Crash in e_addressbook_view_delete_selection() (Milan Crha)
+	Bug 692361 - e_cal_client_get_free_busy() broken (Milan Crha)
+	Bug 746276 - e_client_cache_get_client_sync() gets stuck (Milan Crha)
+	Bug 751225 - Spell check selection changes font settings to initial settings (Tomas Popela)
+
+Miscellaneous:
+	Editor - Width limits are wrong for lists when switching composer modes (Tomas Popela)
+	EHTMLEditor - Improve lists formats (Tomas Popela)
+	EHTMLEditorSelection - Blocks are wrongly wrapped when the last character is space (Tomas Popela)
+	EHTMLEditorSelection - Alignment cannot be set back to left in list (Tomas Popela)
+	EHTMLEditorSelection - Width limit is set when indenting block (Tomas Popela)
+	EHTMLEditorSelection - Fix indentation (Tomas Popela)
+	EHTMLEditorSelection - Alignment could be wrong (Tomas Popela)
+	EHTMLEditorSelection - Current block format is wrong for nested lists (Tomas Popela)
+	EHTMLEditorSelection - Fix the change of a list formatting to another list formatting (Tomas Popela)
+	EHTMLEditorSelection - Fix the formatting change from list to block (Tomas Popela)
+	EHTMLEditorSelection - Roman Numeral and Alphabetical list formats are not detected (Tomas Popela)
+	EHTMLEditorSelection - Copy&pasting quoted content in plain text composer preserves formatting (Tomas Popela)
+	EHTMLEditorView - Improve debug output of undo/redo history events (Tomas Popela)
+	EHTMLEditorView - Remove debug outputs committed by mistake (Tomas Popela)
+	EHTMLEditorView - Fix appending a text to the link (Tomas Popela)
+	EHTMLEditorView - Current block is removed when Backspace is pressed on its end (Tomas Popela)
+	EHTMLEditorView - If deleting quoted content, other content can be deleted as well (Tomas Popela)
+	EHTMLEditorView - Fix indentation (Tomas Popela)
+	EHTMLEditorView - Fix leaks discovered by Valgrind (Tomas Popela)
+	EHTMLEditorView - Allow links to be wrapped on any character, not just on dash (Tomas Popela)
+	EHTMLEditorView - Inserting space after the link appends it to the link (Tomas Popela)
+	EHTMLEditorView - Pasting content from message source into plain text composer preserves formatting (Tomas Popela)
+	Ignore "Service unavailable" errors in Send/Receive routines (Milan Crha)
+	Show "No proxy" instead of "None" in the proxy editor (Milan Crha)
+	Correct widget positions after the change for bug 751594 (Milan Crha)
+	Composer - Leave copy&pasting inside the composer on WebKit (Tomas Popela)
+	[Win32] Update some of the libraries to more recent versions (Milan Crha)
+	Fixed typo in string (Daniel Mustieles)
+	Do not create Contacts source for GMail accounts (Milan Crha)
+	Updated Spanish translation (Daniel Mustieles)
+	All-day event covers two days in the mini-calendar (Milan Crha)
+
+Translations:
+	Daniel Mustieles (es)
+	Gábor Kelemen (hu)
+	Balázs Úr (hu)
+	Pedro Albuquerque (pt)
+	Matej Urbančič (sl)
+	Dušan Kazik (sk)
+	Daniel Mustieles (es)
+	Anders Jonsson (sv)
+
 Evolution 3.17.3 2015-06-22
 ---------------------------
 
-- 
2.15.0

