From cb29d46f8baf256306deaf976b6cae6128ff9508 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Thu, 13 Aug 2015 19:49:44 +0200
Subject: [PATCH 0944/1800] Bug 686363 - Do not canon "From " in the
 composer-generated mails

---
 composer/e-msg-composer.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/composer/e-msg-composer.c b/composer/e-msg-composer.c
index 0324743b75..d5134d2fe6 100644
--- a/composer/e-msg-composer.c
+++ b/composer/e-msg-composer.c
@@ -702,10 +702,6 @@ composer_add_quoted_printable_filter (CamelStream *stream)
 	filter = camel_mime_filter_basic_new (CAMEL_MIME_FILTER_BASIC_QP_ENC);
 	camel_stream_filter_add (CAMEL_STREAM_FILTER (stream), filter);
 	g_object_unref (filter);
-
-	filter = camel_mime_filter_canon_new (CAMEL_MIME_FILTER_CANON_FROM);
-	camel_stream_filter_add (CAMEL_STREAM_FILTER (stream), filter);
-	g_object_unref (filter);
 }
 
 /* Helper for composer_build_message_thread() */
-- 
2.15.0

