From 7c4ee3086529c31cf39c42796f11b534aef2c826 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 14 Sep 2015 11:26:59 +0200
Subject: [PATCH 1019/1800] NEWS update for 3.17.92

---
 NEWS | 48 ++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/NEWS b/NEWS
index f54d911c24..dd45646b4a 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,51 @@
+Evolution 3.17.92 2015-09-14
+----------------------------
+
+Bug Fixes:
+	Bug 675154 - Provide an application menu (Milan Crha)
+	Bug 754329 - Runtime warning when pressing Delete at the end of paragraph (Milan Crha)
+	Bug 661886 - Avoid vertical scrolling in the filtering rules editor dialog (Milan Crha)
+
+Miscellaneous:
+	[Win32] Miscellaneous build changes (Milan Crha)
+	Extend ECanvas cursor positioning routine (Milan Crha)
+	Crash under e_mail_reader_destroy_menu_idle_cb() (Milan Crha)
+	Don't wrap on Unicode's non-breaking space character (Tomas Popela)
+	EHTMLEditorSelection - Last word that could fit on one line is put on the following line (Tomas Popela)
+	EHTMLEditorSelection - Avoid wrong wrapping in some cases (Tomas Popela)
+	EHTMLEditorSelection - Remove the GObject data after using it to avoid possible mis-function (Tomas Popela)
+	EHTMLEditorSelection - Correct wrapping when deleting characters in quoted content (Tomas Popela)
+	EHTMLEditorSelection - Remove trailing spaces (Tomas Popela)
+	EHTMLEditorSelection - Correctly wrap text nodes around selection markers (Tomas Popela)
+	EHTMLEditorSelection - Just first line in block is wrapped (Tomas Popela)
+	EHTMLEditorSelection - Dash is a valid character for wrapping only if it's not surrounded by spaces (Tomas Popela)
+	EHTMLEditorSelection - Correctly wrap text nodes around selection markers (Tomas Popela)
+	EHTMLEditorSelection - If copy/pasting in composer WebKit wraps content in SPAN (Tomas Popela)
+	EHTMLEditorUtils - Add argument where we can specify into what level the node will be split (Tomas Popela)
+	EHTMLEditorView - Try to merge adjacent blockquotes when editing quoted text (Tomas Popela)
+	EHTMLEditorView - Avoid double quotes when deleting a text inside quoted content (Tomas Popela)
+	EHTMLEditorView - Modifying the text of a smiley should convert the smiley to a text node (Tomas Popela)
+	EHTMLEditorView - Bringing anchor from quoted text to body with Backspace doesn't work (Tomas Popela)
+	EHTMLEditorView - Avoid plain text quote characters to be inserted in HTML mode (Tomas Popela)
+	EHTMLEditorView - Undo/Redo history is wrong after redoing UNQUOTE operation (Tomas Popela)
+	EHTMLEditorView - Avoid multiple unnecessary spell checking (Tomas Popela)
+
+Translations:
+	Dušan Kazik (sk)
+	Piotr Drąg (pl)
+	Pedro Albuquerque (pt)
+	Fran Dieguez (gl)
+	Gábor Kelemen (hu)
+	Daniel Mustieles (es)
+	Baurzhan Muftakhidinov (kk)
+	Changwoo Ryu (ko)
+	Christian Kirbach (de)
+	Aurimas Černius (lt)
+	Kjartan Maraas (nb)
+	Muhammet Kara (tr)
+	Alexandre Franke (fr)
+	Anders Jonsson (sv)
+
 Evolution 3.17.91 2015-08-31
 ----------------------------
 
-- 
2.15.0

